crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{
+    c8y::{converter::CumulocityConverter, http_proxy::JwtAuthHttpProxy},
+    mapping::{
+        component::TEdgeComponent, mapper::create_mapper, operations::Operations,
+        size_threshold::SizeThreshold,
+    },
+};
+
+use agent_interface::topic::ResponseTopic;
+use async_trait::async_trait;
+use mqtt_channel::{Config, TopicFilter};
+use tedge_config::{
+    ConfigSettingAccessor, DeviceIdSetting, DeviceTypeSetting, MqttPortSetting, TEdgeConfig,
+};
+use tracing::{info, info_span, Instrument};
+
+use super::topic::C8yTopic;
+
+const CUMULOCITY_MAPPER_NAME: &str = "tedge-mapper-c8y";
+
+pub struct CumulocityMapper {}
+
+impl CumulocityMapper {
+    pub fn new() -> CumulocityMapper {
+        CumulocityMapper {}
+    }
+
+    pub fn subscriptions(operations: &Operations) -> Result<TopicFilter, anyhow::Error> {
+        let mut topic_filter = TopicFilter::new(ResponseTopic::SoftwareListResponse.as_str())?;
+        topic_filter.add(ResponseTopic::SoftwareUpdateResponse.as_str())?;
+        topic_filter.add(C8yTopic::SmartRestRequest.as_str())?;
+        topic_filter.add(ResponseTopic::RestartResponse.as_str())?;
+
+        for topic in operations.topics_for_operations() {
+            topic_filter.add(&topic)?
+        }
+
+        Ok(topic_filter)
+    }
+
+    pub async fn init_session(&mut self) -> Result<(), anyhow::Error> {
+        info!("Initialize tedge sm mapper session");
+        let operations = Operations::try_new("/etc/tedge/operations", "c8y")?;
+        let mqtt_topic = Self::subscriptions(&operations)?;
+        let config = Config::default()
+            .with_session_name(CUMULOCITY_MAPPER_NAME)
+            .with_clean_session(false)
+            .with_subscriptions(mqtt_topic);
+        mqtt_channel::init_session(&config).await?;
+        Ok(())
+    }
+
+    pub async fn clear_session(&mut self) -> Result<(), anyhow::Error> {
```

Comment:
```suggestion
    pub async fn clean_session(&mut self) -> Result<(), anyhow::Error> {
```
I see that you've kept it consistent with the `mqtt_channel::clear_session` API. I wish it also named `clean_session` to better reflect the MQTT protocol contract.